db/migrate/20220915132547_add_dominant_color_to_uploads.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class AddDominantColorToUploads < ActiveRecord::Migration[7.0]
+  def change
+    add_column :uploads, :dominant_color, :text, limit: 6, null: true
+    add_index :uploads, :dominant_color, where: "dominant_color IS NULL"
```

Comment:
```suggestion
    add_index :uploads, :id, where: "dominant_color IS NULL"
```

I'm not sure if indexing on `dominant_color` is meaningful considering that there is a `domainant_color IS NULL` filter. For the query this index is added for, I believe indexing on `id` will be more useful.

Reply:
This is just an optimisation for the backfill, it makes it trivially cheap to grab 100 uploads without a dominant_color. I would chuck created date as well in this index though and grab newest first in the backfill.

Reply:
> This is just an optimisation for the backfill, it makes it trivially cheap to grab 100 uploads without a dominant_color. 

Yup I know that which is why I suggested indexing on `id` instead since the query `orders id DESC`.

Reply:
actually looks like you are using reverse id, but I am not sure you are getting id "for free" in the index, I think you need to cover it?

Reply:
Ah, nice catch - thanks. Updated to `id`, and confirmed the backfill query leans on the index:

```
# EXPLAIN ANALYSE SELECT FROM uploads WHERE dominant_color IS NULL ORDER BY id desc;

---------------------------------------------------------------------------------------------------------------------------------------------
 Index Only Scan Backward using index_uploads_on_id on uploads  (cost=0.14..8.16 rows=1 width=4) (actual time=0.033..0.105 rows=207 loops=1)
   Heap Fetches: 124
 Planning Time: 0.763 ms
 Execution Time: 0.134 ms
(4 rows)